roles/maps/tasks/download_large_file.yml

```diff
@@ -1,57 +1,79 @@
+# * We will work toward supporting torrents, which may be faster than http.
+# * Aria2c will hopefully perform checksums in-memory (even if via disk cache)
+#   during download. This will be faster than a full checksum after download,
+#   which definitely requires disk read for large files.
+# * Once files are fully downloaded and moved out of the temp directory, they
+#   will not be downloaded on subsequent role runs. (This relates to "Updating
+#   data" below)
+#
+# Integrity: Aria2 will perform a sha256 checksum on each downloaded chunk of
+# the file.
+#
+# Security: We will NOT commit sha256 checksums to the repository at this time,
+# but that option remains open if the concern comes up. The best option may be
+# to get the checksum of the meta4 file (and maybe the torrent file?)
+#
+# Consistent versions: Data file versions may correspond to each other (e.g.
+# the "current" vector map should ideally match the search database). This can
+# be handled by putting a date in the file names of the data files, perhaps
+# committed to the repository. However, at this time this concern is handled
+# upstream of this file.
+#
+# Updating data: As of Dec 2025 this concern is out of scope. We assume that
+# users will either figure it out on their own, or reinstall IIAB from scratch.
+# That said, assuming we have dates in file names, this may be supported de
+# facto, but we will not go out of our way to free up room by deleting the
+# outdated files. (Perhaps adding automatic deletion will not be so hard once
+# the time comes)
 #
 # NOTE Already downloaded files show up as "skipped" instead of "ok".  I could
 # fix this but we'll move to the download manager in due time anyway.
 # TODO - files should be owned by www-data
 
-- name: "Check previous sha256sum, if any, for {{ item.src_url }})"
+- name: "Download {{ iiab_map_host_url }}/{{ item }} via .meta4 file"
+  # cd to the temp directory so that the aria2c file, meta4 file, torrent file,
+  # and data file all end up there. Delete any possible remnants of previous
+  # attempts to download the file before, to avoid errors due to renaming
+  # (appending .1, etc)
   shell: |
     set -euo pipefail
 
-    cat {{ dest_sha256_path }} || echo "nope"
-  args:
-    executable: /bin/bash
-  register: current_sha256sum
-  vars:
-    # Where the file's sha256sum is saved
-    dest_sha256_path: "{{ dest_base_path }}/{{ item.filename }}.sha256"
+    cd /library/downloads/maps/
 
-- name: "Download {{ item.src_url }}"
-  # Do `wget -c` in hopes of continuing incomplete downloads on the temp file.
-  # However, if the hash doesn't match afterwards, delete the temp file.
-  # I'm not sure if the `echo` statements will do anything useful for debugging.
-  shell: |
-    set -euo pipefail
+    rm -f {{ item }}*
+
+    wget "{{ iiab_map_host_url }}/{{ item }}.meta4"
+    wget "{{ iiab_map_host_url }}/{{ item }}.torrent"
+    aria2c \
+        --metalink-file="{{ item }}.meta4" \
```

Comment:
So @chapmanjacobd  I *thought* I tried it as you suggested (off-channel), with the url to the remote metalink file being the non-parameter argument. Is that actually what I want? And skip the wget above?

Reply:
Yeah that's a good question I would say leave the torrent file off for now and only specify the meta4 file. The meta4 file actually links to the torrent file but I don't know if it actually uses it as long as a HTTP mirror is working. We'll need to play around with it to discover its behavior

Reply:
Yes. I would assume that it would prefix relative paths with the base that it loads the data from. If loaded locally then it wouldn't know where to get the .torrent file except the same local directory. If loaded from a remote, prefix the remote path that is the parent of the meta4 file